### PR TITLE
Issue #51 - Adding /third_party/

### PR DESCRIPTION
**Pull request by joakime**
Initial import of Dandelion
- From https://github.com/dandelion/dandelion.git
- Commit 5daa05e7566a3a7753f13255ea8e5ad68be684e3
- Link dandelion/dandelion@5daa05e7566a3a7753f13255ea8e5ad68be684e3

Initial import of Spring Petclinic
- From https://github.com/spring-projects/spring-petclinic.git
- Commit 9e6a8caf31844c46c5d56b6d13fe8de933a20ac4
- Link spring-projects/spring-petclinic@9e6a8caf31844c46c5d56b6d13fe8de933a20ac4

Applying pull request dandelion/dandelion#116
changing groupId's to com.google.cloud.runtime.jetty.third_party
Updating maven build for third_party

### CODE REVIEW

**Comment by meltsufin** [comment]
@joakime Did you see the comment from the maintainer of Spring Petclinic that they will be removing Dandillion altogether from the project? https://github.com/spring-projects/spring-petclinic/issues/171

**Comment by joakime** [comment]
Do we want to wait for spring-petclinic to fix this? or just use the PR as-is and move the testing forward now, vs much much later?

Lets use these from the PR, then if in some point in the future spring-petclinic is updated (great!) we can  remove these /third_party/ entries in favor of the official ones.

**Comment by joakime** [comment]
@meltsufin is this ok to merge yet?

File: third_party/pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+>
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.google.cloud.runtimes</groupId>
```
Comment (meltsufin) [inline third_party/pom.xml]: What's the goal of connecting it to the parent? Maybe connect to the `/test/pom.xml` instead, if anything at all?

**Comment by joakime** [comment]
@meltsufin responded to review & updated branch to allow easier merging.

File: third_party/pom.xml (hunk 1 above)
Comment (joakime) [inline third_party/pom.xml]: maven convention over configuration.
child modules reference their parent.

proposing an alternate approach then.
Drop `third_party/pom.xml` and just modify the parent `[jetty-runtime]/pom.xml` to add ...

``` xml
    <module>third_party/dandelion</module>
    <module>third_party/spring_petclinic</module>
```

With those specific poms not referencing the `[jetty-runtime]/pom.xml` parent pom.

We should make an effort to modify the `third_party/dandelion/pom.xml` and `third_party/spring_petclinic/pom.xml` to not deploy (snapshots or releases).
These modifications should appear as a separate PR to be very clear about the changes.

We don't need to for the testcases from Issue #33 anyway (we can rely on reactor and/or local repository just fine)

**Comment by joakime** [comment]
@meltsufin bump

**Comment by meltsufin** [comment]
We had a discussion internally today about importing third party apps into our repo. We're thinking of using a fork and a git submodule, or something similar, instead. We'll get back to you on this shortly.

**Comment by meltsufin** [comment]
@joakime What do you think of using submodules for dandelion and petclinic? Take a look at the experimental branch @aslo put together: https://github.com/GoogleCloudPlatform/jetty-runtime/tree/issue-51--submodules.
Can you fork the two repos under the Jetty account, and then point to the modified forks using submodules here?

**Comment by joakime** [comment]
We could use git submodules, but you will still need to fork and host those projects somewhere where the fixes are in place.

The fixes:

 * Dandelion patch applied
 * Dandelion version updated to prevent use of repository based version
 * Dandelion testing disabled (not compatible with java 8)
 * Dandelion's maven-deploy-plugin is skipped
 * Spring petclinic patched to use updated/patched dandelion
 * Spring petclinic maven-deploy-plugin is skipped

That's just want I've identified so far.

Other work that's in progress that could affect this list...
I'm working on Issue #33 and hope to be able to reuse the `[spring_petclinic]/src/test/` content to perform remote testing against the two different `spring_petclinic` deployments (as-is, and google integrated)

**Comment by meltsufin** [comment]
Note that the Pet Clinic project has just removed Dandelion: https://github.com/spring-projects/spring-petclinic/issues/171#event-816510176

**Comment by jmcc0nn3ll** [comment]
closing for the time being, will bring back as needed down the road

**Pull request closed without merging**